modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js

```diff
@@ -536,12 +534,20 @@ AUI.add(
 				},
 
 				_destroyBanner() {
-					var instance = this;
+					const instance = this;
+
+					const toast = document.getElementById(TOAST_ID);
+
+					const toastRootElement = toast?.parentElement;
 
-					if (Lang.isFunction(instance._banner.destroy)) {
-						instance._banner.destroy();
+					instance._banner.destroy();
+
+					if (toastRootElement) {
```

Comment:
I remember this rootElement was necessary for stacking toasts inside the window. It can inhibit other toasts to be stacked above session one. Could you do this test?

Reply:
For stacking to work, [container](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/toast/commands/OpenToast.es.js#L39) needs to be preserved. But, `toastRootElement` here is the `<div class="mb-3">`, added by [this code](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/toast/commands/OpenToast.es.js#L68-L69) whenever a toast opens. In the toast component, we are [removing it on close](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/toast/commands/OpenToast.es.js#L121). However, when we are destroying the component, we are not invoking `onClose`, only [unmounting the component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.tsx#L70). That way, the div stays behind. You can see this in the 'Before PR' gif.

Reply:
Nice! Thanks for the explanations :)